cmd/backup-manager/app/backup/manager.go

```diff
@@ -114,42 +114,65 @@ func (bm *BackupManager) performBackup(backup *v1alpha1.Backup, db *sql.DB) erro
 	}
 	glog.Infof("cluster %s %s is %s", bm, constants.TikvGCVariable, oldTikvGCTime)
 
-	err = bm.setTikvGCLifeTime(db, constants.TikvGCLifeTime)
+	oldTikvGCTimeDuration, err := time.ParseDuration(oldTikvGCTime)
 	if err != nil {
-		glog.Errorf("cluster %s set tikv GC life time to %s failed, err: %s", bm, constants.TikvGCLifeTime, err)
+		glog.Errorf("cluster %s parse old %s failed, err: %s", bm, constants.TikvGCVariable, err)
 		return bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
 			Type:    v1alpha1.BackupFailed,
 			Status:  corev1.ConditionTrue,
-			Reason:  "SetTikvGCLifeTimeFailed",
+			Reason:  "ParseOldTikvGCLifeTimeFailed",
 			Message: err.Error(),
 		})
 	}
-	glog.Infof("set cluster %s %s to %s success", bm, constants.TikvGCVariable, constants.TikvGCLifeTime)
-
-	backupFullPath, err := bm.dumpTidbClusterData()
+	tikvGCTimeDuration, err := time.ParseDuration(constants.TikvGCLifeTime)
 	if err != nil {
-		glog.Errorf("dump cluster %s data failed, err: %s", bm, err)
+		glog.Errorf("cluster %s parse default %s failed, err: %s", bm, constants.TikvGCVariable, err)
 		return bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
 			Type:    v1alpha1.BackupFailed,
 			Status:  corev1.ConditionTrue,
-			Reason:  "DumpTidbClusterFailed",
+			Reason:  "ParseDefaultTikvGCLifeTimeFailed",
 			Message: err.Error(),
 		})
 	}
-	glog.Infof("dump cluster %s data to %s success", bm, backupFullPath)
+	if oldTikvGCTimeDuration < tikvGCTimeDuration {
+		err = bm.setTikvGCLifeTime(db, constants.TikvGCLifeTime)
+		if err != nil {
+			glog.Errorf("cluster %s set tikv GC life time to %s failed, err: %s", bm, constants.TikvGCLifeTime, err)
+			return bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
+				Type:    v1alpha1.BackupFailed,
+				Status:  corev1.ConditionTrue,
+				Reason:  "SetTikvGCLifeTimeFailed",
+				Message: err.Error(),
+			})
+		}
+		glog.Infof("set cluster %s %s to %s success", bm, constants.TikvGCVariable, constants.TikvGCLifeTime)
+	}
 
-	err = bm.setTikvGCLifeTime(db, oldTikvGCTime)
+	backupFullPath, err := bm.dumpTidbClusterData()
 	if err != nil {
-		glog.Errorf("cluster %s reset tikv GC life time to %s failed, err: %s", bm, oldTikvGCTime, err)
+		glog.Errorf("dump cluster %s data failed, err: %s", bm, err)
 		return bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
 			Type:    v1alpha1.BackupFailed,
 			Status:  corev1.ConditionTrue,
-			Reason:  "ResetTikvGCLifeTimeFailed",
+			Reason:  "DumpTidbClusterFailed",
 			Message: err.Error(),
 		})
 	}
-	glog.Infof("reset cluster %s %s to %s success", bm, constants.TikvGCVariable, oldTikvGCTime)
+	glog.Infof("dump cluster %s data to %s success", bm, backupFullPath)
 
+	if oldTikvGCTimeDuration < tikvGCTimeDuration {
```

Comment:
Should we try to reset this with backoff? @onlymellb Since this is an important parameter, we should ensure this is reset as possible. We don't have to do it in this PR.

Reply:
@tennix If this value is not set, many TiDBs cannot be backed up. For example, the value set in our production environment is: 10m. This value cannot back up TiDB data. Only the setting is large enough to back up data.

Reply:
What I'm trying to say is when the backup is done, we need to ensure the gc lifetime is reset to the original value. Currently, this seems to be done only once and print error logs when fails. We can retry with backoff if the value is still what we set during backup for example 72h.

Reply:
This is indeed a hidden question, and I agree with you very much, this possibility is very high.

Reply:
I think this is necessary, we can use this [function](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L290-L301) to do exponential backoff